src/main/java/org/generationcp/ibpworkbench/germplasm/pedigree/CreatePedigreeGraph.java

```diff
@@ -138,36 +139,34 @@ private void addNode(final GermplasmPedigreeTreeNode node, final int level) {
 			final String leafNodeGIDRoot = this.createNodeTextWithFormatting(node);
 			this.graphVizUtility.addln(leafNodeGIDRoot + ";");
 		}
-
 		for (final GermplasmPedigreeTreeNode parent : node.getLinkedNodes()) {
-			final boolean isFemaleParent = parent.getGermplasm().getGid().equals(node.getGermplasm().getGpid1())
-				&& node.getGermplasm().getGnpgs() >= 2;
-			final boolean isDerivative = node.getGermplasm().getGnpgs().equals(-1);
 			final String leafNodeGID = this.createNodeTextWithFormatting(parent);
 			final String parentNodeGID = this.createNodeTextWithFormatting(node);
 
 			if (level == 1) {
 				final String leafNodeGIDRoot = this.createNodeTextWithFormatting(node);
-				this.graphVizUtility.addln(leafNodeGID + "->" + leafNodeGIDRoot + this.getArrowStyleString(isFemaleParent, isDerivative));
+				this.graphVizUtility.addln(leafNodeGID + "->" + leafNodeGIDRoot + this.getArrowStyleString(node.getGermplasm(), parent.getGermplasm()));
 			} else {
-				this.graphVizUtility.addln(leafNodeGID + "->" + parentNodeGID + this.getArrowStyleString(isFemaleParent, isDerivative));
+				this.graphVizUtility.addln(leafNodeGID + "->" + parentNodeGID + this.getArrowStyleString(node.getGermplasm(), parent.getGermplasm()));
 			}
 
 			this.addNode(parent, level + 1);
 		}
 	}
 
-	String getArrowStyleString(final boolean isFemaleParent, final boolean isDerivative) {
-		if(isFemaleParent) {
-			return " [color=\"RED\", arrowhead=\"odottee\"];";
-		} else if(!isDerivative) {
-			return " [color=\"BLUE\", arrowhead=\"veeodot\"];";
-		} else {
-			return  ";";
+	public String getArrowStyleString(Germplasm nodeGermplasm, Germplasm parentGermplasm) {
+		if (nodeGermplasm.getGnpgs() != null && nodeGermplasm.getGnpgs() >= 2 ) {
+			if (parentGermplasm.getGid().equals(nodeGermplasm.getGpid1())) {
+				//Female Parent
+				return " [color=\"RED\", arrowhead=\"odottee\"];";
+			} else if (parentGermplasm.getGid().equals(nodeGermplasm.getGpid2())) {
```

Comment:
How does this handle polycrosses? will the 3rd, 4th, 5th etc parent also be coloured blue?

Reply:
Test using germplsm with breeding method Three-way cross (Pedigree level upto 5). Compared to master and the result is the same. If the node gpid2 is equal to current gid (parent), the color will be blue, if equal to gpid1 the color will be red otherwise black